src/api/handlers/twilio/postReply.js

```diff
@@ -1,15 +1,157 @@
+    .where({ patientPhoneNumber, appointmentIsConfirmed: 0 });
+}
+
+export async function handlePostReply(patientPhoneNumber, messageFromPatient) {
+  let ourResponse;
+
+  if (!messageFromPatient || messageFromPatient === "") {
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not process your response. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
+
+  const lowercaseMessageFromPatient = messageFromPatient.toLowerCase();
+
+  let appointments;
+  try {
+    appointments = await getAppointments(patientPhoneNumber);
+  } catch (error) {
+    // Can't store reply without appointmentId
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not find an appointment for you. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
+
+  if (!appointments || appointments.length === 0) {
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not find an appointment for you, or your appointment is already confirmed. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
+
+  const trx = await db.transaction();
+  const convertedReply = convertReply(lowercaseMessageFromPatient);
 
-  const reply = getMessageResponse(replyText);
+  if (!convertedReply) {
+    ourResponse = getMessageResponse(
+      "We are sorry, we could not confirm your appointment. Please call Sanctuary Refugee Health Centre (Dr. Michael Stephenson) at 226-336-1321"
+    );
+    return ourResponse;
+  }
 
+  try {
+    switch (lowercaseMessageFromPatient) {
+      case "yes":
+        for (const appointment of appointments) {
+          await db("replies")
+            .insert({
+              phoneNumber: patientPhoneNumber,
+              body: messageFromPatient,
+              appointmentId: appointment.appointmentId,
+              time: new Date(),
+            })
+            .transacting(trx);
+          await db("appointments")
+            .where({ appointmentId: appointment.appointmentId })
+            .update({
+              appointmentIsConfirmed: true,
+            })
+            .transacting(trx);
+        }
+        break;
+      case "no":
+        for (const appointment of appointments) {
+          await db("replies")
+            .insert({
+              phoneNumber: patientPhoneNumber,
+              body: messageFromPatient,
+              appointmentId: appointment.appointmentId,
+              time: moment().format("YYYY-MM-DD HH:mm:ss"),
+            })
+            .transacting(trx);
+          await db("appointments")
+            .where({ appointmentId: appointment.appointmentId })
+            .update({
+              appointmentIsConfirmed: false,
+            })
+            .transacting(trx);
+        }
+        break;
+      // Handle interpreter?
+      default:
```

Comment:
How about handling the case when the user reply did not recognize as one of the cases above (yes, no or interpreter) by replying with a message asking the user to reply with a valid response. something like:

> We are sorry, your response not valid. Could you please send (Yes) to confirm the appointment, (No) to decline the appointment, or (Interpreter) to request an interpretation aid with the appointment.

Reply:
That case is handled above with a message already. We can change it of course. Interpreter needs to be handled still as part of a separate ticket and we have no translations.

Reply:
Ah right! sounds good 👍